boards/DIM/Src/App/App_SevenSegDisplays.c

```diff
@@ -77,5 +81,40 @@ ErrorCode App_SevenSegDisplays_SetHexDigits(
             seven_seg_displays->displays[i], hex_digit);
     }
 
-    return ERROR_CODE_OK;
+    if (seven_seg_displays->set_value_callback != NULL)
+    {
+        seven_seg_displays->set_value_callback();
+    }
+
+    return EXIT_CODE_OK;
+}
+
+ExitCode App_SevenSegDisplays_SetUnsignedBase10Value(
+    const struct SevenSegDisplays *const seven_seg_displays,
+    uint32_t                             value)
+{
+    if (value > pow(10, NUM_SEVEN_SEG_DISPLAYS) - 1)
+    {
+        return EXIT_CODE_INVALID_ARGS;
+    }
+
+    uint8_t digits[NUM_SEVEN_SEG_DISPLAYS];
+
+    // Turn the base-10 value into individual digits. We treat a value of 0
+    // as having 1 digit, which is why num_digits starts counting from 1.
+    for (uint8_t num_digits = 1; num_digits <= NUM_SEVEN_SEG_DISPLAYS;
+         num_digits++)
+    {
+        digits[num_digits - 1] = value % 10;
+        value /= 10;
+
+        if (value == 0)
+        {
+            App_SevenSegDisplays_SetHexDigits(
+                seven_seg_displays, digits, num_digits);
+            break;
+        }
+    }
+
+    return EXIT_CODE_OK;
```

Comment:
Just to be clear, are we displaying all digits/error codes in base 10? Just want to make sure we're OK with giving up the extra values that displaying in hex gives us.

Reply:
Yeah so i was reviewing the specs on what exactly we need to display, and there are

- The SoC: This has to be in base 10 since it goes from 0 to 100
- The error codes follow the scheme [One digit for Board ID] [Two digits for Error Code], leaving 99 error codes per board which should be more than enough. I feel like we should change `App_SevenSegDisplays_SetHexDigits` to a static function because I don't see any use-case (arguably we could even just remove it but I'd leave it in for now).

Reply:
Actually I thought it is strictly better to hide `SetHexDigits` so I turned it into a static function d7dc5c1af2e46993ccda13e0673bdac2190e7b10

Reply:
Uggh the existing tests for `SetHexDigits` can't compile anymore because well, it's a private function now. What should we do in this case @garethellis0 ?

Reply:
Leave them public?

Reply:
Didn't want to leave them public if we're not using it but I guess it's more important to test the function